From 6e6aad75a8f03ba46307059d64417fd0d5bfac1b Mon Sep 17 00:00:00 2001
From: Luca Stefani <luca.stefani.ge1@gmail.com>
Date: Fri, 19 Oct 2018 11:57:15 +0200
Subject: [PATCH 5/6] Merge android-9.0.0_r12

commit 5fef3204fe5011804473267426daf12377e9c755
Merge: 03cb53a17 866374163
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Tue Jul 3 03:04:08 2018 +0000

    Snap for 4874588 from 866374163a1164e1113a4d2e9e0453d68a06dfa3 to pi-dr1-release

    Change-Id: Iac0482218f294d930b24f88b7f31547aeb27538b

commit 866374163a1164e1113a4d2e9e0453d68a06dfa3
Merge: 03cb53a17 9893d6d0f
Author: Tom Cherry <tomcherry@google.com>
Date:   Mon Jul 2 16:02:52 2018 +0000

    Merge "Reland "Remove a check for AIDs in the OEM range."" into pi-dev

commit 9893d6d0ff2be5a6cdfb944df17b394678a7e340
Author: Tom Cherry <tomcherry@google.com>
Date:   Fri Jun 29 10:39:43 2018 -0700

    Reland "Remove a check for AIDs in the OEM range."

    There may be vendor extensions to the system image which would need to
    use these IDs.

    This reverts commit 41986a013a10c6f79822002800a9afe90efba4f2.

    Bug: 110856218
    Test: these tests

Change-Id: I688b8fbfdd329b3e4cf6658abbba5cbcbd32ddc4
---
 tests/grp_pwd_test.cpp | 18 ++++++++++++++++--
 1 file changed, 16 insertions(+), 2 deletions(-)

diff --git a/tests/grp_pwd_test.cpp b/tests/grp_pwd_test.cpp
index fa8a662f2..6316387f0 100644
--- a/tests/grp_pwd_test.cpp
+++ b/tests/grp_pwd_test.cpp
@@ -275,7 +275,14 @@ TEST(pwd, getpwent_iterate) {
       EXPECT_STREQ("/data", pwd->pw_dir) << "pwd->pw_uid: " << pwd->pw_uid;
     }
 
-    EXPECT_EQ(0U, uids.count(pwd->pw_uid)) << "pwd->pw_uid: " << pwd->pw_uid;
+    // TODO(b/27999086): fix this check with the OEM range
+    // If OEMs add their own AIDs to private/android_filesystem_config.h, this check will fail.
+    // Long term we want to create a better solution for OEMs adding AIDs, but we're not there
+    // yet, so therefore we do not check for uid's in the OEM range.
+    if (!(pwd->pw_uid >= 2900 && pwd->pw_uid <= 2999) &&
+        !(pwd->pw_uid >= 5000 && pwd->pw_uid <= 5999)) {
+      EXPECT_EQ(0U, uids.count(pwd->pw_uid)) << "pwd->pw_uid: " << pwd->pw_uid;
+    }
     uids.emplace(pwd->pw_uid);
   }
   endpwent();
@@ -518,7 +525,14 @@ TEST(grp, getgrent_iterate) {
     EXPECT_STREQ(grp->gr_name, grp->gr_mem[0]) << "grp->gr_gid: " << grp->gr_gid;
     EXPECT_TRUE(grp->gr_mem[1] == NULL) << "grp->gr_gid: " << grp->gr_gid;
 
-    EXPECT_EQ(0U, gids.count(grp->gr_gid)) << "grp->gr_gid: " << grp->gr_gid;
+    // TODO(b/27999086): fix this check with the OEM range
+    // If OEMs add their own AIDs to private/android_filesystem_config.h, this check will fail.
+    // Long term we want to create a better solution for OEMs adding AIDs, but we're not there
+    // yet, so therefore we do not check for gid's in the OEM range.
+    if (!(grp->gr_gid >= 2900 && grp->gr_gid <= 2999) &&
+        !(grp->gr_gid >= 5000 && grp->gr_gid <= 5999)) {
+      EXPECT_EQ(0U, gids.count(grp->gr_gid)) << "grp->gr_gid: " << grp->gr_gid;
+    }
     gids.emplace(grp->gr_gid);
   }
   endgrent();
-- 
2.17.1

